src/machine/state/mod.rs

```diff
@@ -16,6 +17,49 @@ pub struct State {
 
 impl State {
     pub fn new() -> Self {
```

Comment:
Maybe we should better write a constructor `pub fn with_memory(bytes: u64) -> Self`. Or we use the builder pattern?!